lib/tsParser.js

```diff
@@ -13,26 +13,49 @@ marked.setOptions({
   }
 });
 
-var app = new typedoc.Application({
-  mode: 'Modules',
-  logger: 'console',
-  target: 'ES6',
-  module: 'CommonJS',
-  experimentalDecorators: true,
-  includeDeclarations: true,
-});
-
-function TSParser(filePaths) {
+function TSParser(filePaths, config) {
   this.filePaths = filePaths;
   this.sections = [];
   this.constructs = [];
+  config = config || {};
+  var options = {
+    mode: 'Modules',
+    logger: 'console',
+    target: config.tstarget || 'ES6',
+    module: 'CommonJS',
+    experimentalDecorators: true,
+    includeDeclarations: true,
+  };
+  if (config.tsconfig) {
+    options.tsconfig = config.tsconfig;
+  }
+  this.app = new typedoc.Application(options);
+}
+
+// Override typedoc.Application.convert() to get errors
+function convert(app, src) {
+  app.logger.writeln(
+    "Using TypeScript %s from %s",
+    app.getTypeScriptVersion(),
+    app.getTypeScriptPath()
+  );
+
+  var result = app.converter.convert(src);
+  if (result.errors && result.errors.length) {
+    app.logger.diagnostics(result.errors);
+    if (app.ignoreCompilerErrors) {
+      app.logger.resetErrors();
+    } 
+  } 
+  return result;
 }
 
 TSParser.prototype.parse = function() {
-  var project = app.convert(this.filePaths);
+  var result = convert(this.app, this.filePaths);
+  var project = result.project;
   
-  if (!project) {
-    console.log('Could not create project for file: ' + this.filename);
+  if (result.errors && result.errors.length) {
+    console.error('Could not create project for file: ' + result.errors);
```

Comment:
nitpick: maybe keep the `this.filename` in log information like the old code.

Reply:
In fact, the `result` does not have `filename` property.